app/src/main/java/ru/practicum/android/diploma/filters/presentation/ui/FiltersFragment.kt

```diff
@@ -79,21 +119,50 @@ class FiltersFragment : Fragment() {
         }
         binding.btnChoose.setOnClickListener {
             vm.saveFilters(
-                binding.locationEditText.text.toString(),
-                binding.industryEditText.text.toString(),
                 binding.salaryEditText.text.toString(),
                 isChecked,
             )
+            setFragmentResult(
+                SET_FILTERS_KEY,
+                bundleOf()
+            )
+            findNavController().navigateUp()
         }
 
         binding.btnRemove.setOnClickListener {
             vm.removeFilters()
             binding.locationEditText.text?.clear()
+            binding.industryEditText.text?.clear()
         }
 
         vm.observeState().observe(viewLifecycleOwner) {
-            binding.locationEditText.setText(it.location)
-            binding.industryEditText.setText(it.industry)
+            setFragmentResult(
+                COUNTRY_AND_REGION,
+                bundleOf(
+                    COUNTRY_JSON_KEY to it.country,
+                    REGION_JSON_KEY to it.region
+                )
+            )
+            if (!it.country.isNullOrEmpty()) {
+                country = Json.decodeFromString(it.country ?: "")
+                if (!it.region.isNullOrEmpty()) {
+                    region = Json.decodeFromString(it.region ?: "")
+                    val location = country.name + "," + region.name
```

Comment:
🍏 
```suggestion
                    val location = "${country.name}, ${region.name)"
```
https://www.baeldung.com/kotlin/string-templates